localization/pose_estimator/ndt_scan_matcher/ndt_omp/CMakeLists.txt

```diff
@@ -1,70 +1,61 @@
-cmake_minimum_required(VERSION 3.0.2)
+cmake_minimum_required(VERSION 3.5)
 project(ndt_omp)
 
-# -mavx causes a lot of errors!!
-add_definitions(-std=c++14 -msse -msse2 -msse3 -msse4 -msse4.1 -msse4.2)
-set(CMAKE_CXX_FLAGS "-std=c++14 -msse -msse2 -msse3 -msse4 -msse4.1 -msse4.2")
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
 
 # warn performance issue of ndt_omp which is NOT built in release mode
-string(TOUPPER ${CMAKE_BUILD_TYPE} uppercase_CMAKE_BUILD_TYPE)
-if (NOT (${uppercase_CMAKE_BUILD_TYPE} STREQUAL "RELEASE"))
-  message(WARNING
-    "CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} may cause a serious performance degradation. Please configure with -DCMAKE_BUILD_TYPE=Release.")
+if (NOT ("${CMAKE_BUILD_TYPE}" STREQUAL "Release"))
```

Comment:
This will always emit a warning since I think we rarely compile in Release mode. Is this the desired behavior?
How about just always enabling optimization with an unconditional `-O2` flag or similar?

Reply:
@nnmm I guess you could set the release type yourself. But this check is an anti-pattern anyways as it wouldn't work the multi-config builds such as with clion; see https://stackoverflow.com/a/24470998/14409577

But yes, this highlights an important point that I raised here https://github.com/colcon/colcon-cmake/issues/96. For the next standup, we should bring up the default build config

Reply:
I just set the build type to Release, that's what the upstream packages does too.